### PR TITLE
Add file upload callback registration for mavlink FTP server

### PR DESCRIPTION
**Pull request by MatejFranceskin**
This PR adds a method to register file uploaded to Mavlink FTP server callback.
When file is successfully uploaded the callback is called with the file path as parameter.

### CODE REVIEW

**Comment by JonasVautherin** [comment]
Thanks Matej!

How is that different from `upload_async(std::string local_file_path, std::string remote_dir, UploadCallback callback);`? Doesn't the callback serve the same purpose?

**Comment by julianoes** [comment]
@JonasVautherin this is for the server to be notified when something has been uploaded, so it's different, I think.

File: src/plugins/ftp/include/plugins/ftp/ftp.h [hunk 1]
```diff
@@ -139,6 +139,17 @@ class Ftp : public PluginBase {
...
+     * @brief Callback type for new file uploaded by remote.
+     */
+
+    using FileUploadedCallback = std::function<void(std::string)>;
+
+    /**
+     * @brief Register a callback to be triggered when remote uploads a file to FTP server
+     */
+    void register_file_uploaded_callback(FileUploadedCallback callback);
+
```
Comment (julianoes) [inline src/plugins/ftp/include/plugins/ftp/ftp.h:152]: This file is auto-generated from https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/ftp/ftp.proto, you can't change it like that.

**Review by julianoes — CHANGES_REQUESTED** [review]

**Comment by MatejFranceskin** [comment]
After more testing I found some additional problems. QGC mavlink FTP client ends the upload with reset session instead of terminate and I had to find a different way to achive what I needed.
I am closing this since it is not needed anymore and it might cause problems.

**Pull request closed without merging**